src/pages/docs/api/rest-sdk/push-admin.mdx

```diff
@@ -0,0 +1,1553 @@
+|----------|-------------|------|
+| id | unique identifier for the device generated by the device itself | String |
+| <If lang="javascript,nodejs,java,android,php,swift,objc">clientId</If><If lang="ruby">client_id</If> | optional trusted [client identifier](/docs/auth/identified-clients) for the device | String |
+| <If lang="javascript,nodejs,java,android,php,swift,objc">formFactor</If><If lang="ruby">form_factor</If> | form factor of the push device. Must be one of `phone`, `tablet`, `desktop`, `tv`, `watch`, `car` <If lang="javascript,nodejs,java,android,php,ruby,swift,objc">or `embedded`</If><If lang="javascript,nodejs">or `other`</If> | String |
+| <If lang="javascript,nodejs,java,android,php,ruby,swift,objc">metadata</If> | optional metadata object for this device. The metadata for a device may only be set by clients with `push-admin` privileges | <If lang="javascript,nodejs,java,android,swift,objc">Object</If><If lang="php">Array</If><If lang="ruby">Hash</If> |
+| <If lang="javascript,nodejs,java,android,php,ruby,swift,objc">platform</If> | platform of the push device. Must be one of `ios` <If lang="javascript,nodejs,java,android,php,ruby,swift,objc">or `android`</If><If lang="javascript,nodejs">or `browser`</If> | String |
+| <If lang="javascript,nodejs,java,android,php,ruby,swift,objc">deviceSecret</If> | Secret value for the device. | String |
+| push.recipient | push recipient details for this device. See the [REST API push publish documentation](/docs/api/rest-api#message-extras-push) for more details | <If lang="javascript,nodejs,java,android,swift,objc">Object</If><If lang="php">Array</If><If lang="ruby">Hash</If> |
+| push.state | the current state of the push device being either <If lang="javascript,nodejs,java,android,php,ruby,swift,objc">`Active`, `Failing` or `Failed`</If><If lang="javascript,nodejs">`ACTIVE`, `FAILING` or `FAILED`</If> | String |
+| <If lang="java,android,php,swift,objc">push.errorReason</If><If lang="javascript,nodejs">push.error</If><If lang="ruby">push.error_reason</If> | when the device's state is failing or failed, this attribute contains the reason for the most recent failure | [`ErrorInfo`](/docs/api/realtime-sdk/types#error-info) |
+
+### <If lang="javascript,nodejs,java,android,php,objc,swift">PushChannel</If><If lang="ruby">Ably::Models::PushChannel</If> <a id="push-channel" />
+
+A `PushChannel` is a property of a [`RealtimeChannel`](/docs/api/realtime-sdk/channels#properties) or [`RestChannel`](/docs/api/rest-sdk/channels#properties). It provides [push devices](/docs/push) the ability to subscribe and unsubscribe to push notifications on channels.
+
+#### Methods
+
+##### <If lang="javascript,nodejs,java,android,php,ruby,swift,objc">subscribeDevice</If> <a id="subscribe-device" />
+
+<If lang="javascript,nodejs">
+
+`subscribeDevice(): Promise<void>`
+
+</If>
+
+Subscribe your device to the channel's push notifications.
+
+<If lang="javascript,nodejs">
+
+##### Returns
```

Comment:
As with the original. There are lots of the same "returns" section. Even more so in this part of the page

<img width="491" height="696" alt="Image" src="https://github.com/user-attachments/assets/0ee77332-fe71-4744-acb5-29dac332e708" />